### DISCUSSION TITLE
How to extend PrismaClient base functionalities correctly?

### DISCUSSION THREAD

**Discussion by TheVoid-0** [prisma/prisma#11608] [Q&A] [answered]
Hi, thanks for the job done so far!

After a lot of workarounds I finnally managed to put Prisma in a Production API! However I was hoping version  3.9 would solve some of my problems, like the schema path for multiple clients, but it actually ended up breaking my application! Let me explain:

I have Connection class that extends the PrismaClient class, that was needed for my app for compatibility with the rest of the functioning code, and I also thought it would be a good way of extending the base functionalities of PrismaClient, like for example, adding a custom $disconnect() where I can tell my pool manager that the connection is free and etc... But after 3.9 that class stopped working because it somehow has become fully sealed, every property is now readonly and I cannot modify anything, I rolled back to 3.8 and everything is working again.

So I need to ask, is this the desired behavior of the Client? It should not be extended this way? If so, how would be the correct manner to extend the functionality in a OOP way?

Thanks in advance guys, I'm looking forward to the development of prisma!

**Comment by pantharshit00** [comment, marked as answer]
Hello, 

That sound like a bug to me in 3.9. Can you please report the issue here with the code that you are using: https://github.com/prisma/prisma/issues/new?assignees=&labels=kind%2Fbug&template=bug_report.yml

**Reply by TheVoid-0** [reply]
Thanks for the answer. opened #11628

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
